bin/createInhouseSamplesheetFromGS_v2.py

```diff
@@ -0,0 +1,262 @@
+    sys.exit('FATAL ERROR!')
+#
+# Check availability and readability of GS samplesheet.
+#
+for gsSamplesheetFile in glob.iglob(os.path.join(args.genomeScanInputDir, "UMCG_CSV_*.csv.converted")):
+    if os.path.isfile(gsSamplesheetFile) and os.access(gsSamplesheetFile, os.R_OK):
+        logging.info('Found GenomeScan samplesheet: ' + gsSamplesheetFile + '.')
+    else:
+        logging.critical('Either the ' + args.genomeScanInputDir + 'UMCG_CSV_*.csv.converted file is missing or it is not readable.')
+        sys.exit('FATAL ERROR!')
+
+
+# For debugging data structure only:
+#import pprint
+#pp = pprint.PrettyPrinter(indent=4)
+#pp.pprint(gsFilenameDataHashmap)
+#pp.pprint('=====================================================')
+
+#
+# Combine GS samplesheet with original filename information form gsFilenameDataHashmap.
+#
+# GS samplesheet filestructure for samples prepped not at GenomeScan and send in as pools:
+#          GS_ID,         Sample_ID, Pool,   Index1,   Index2
+# 103473-011-001, QXTR_222-Exoom_v1,    1, CGAGGCTG, AGGCTTAG
+#
+## GS samplesheet filestructure for samples prepped at GenomeScan:
+#          GS_ID,                    ID, positie, geslacht, Pool,   Index1,   Index2
+# 103473-011-001, GS_1A-Exoom_v3-123456,     A01,        V,    1, CGAGGCTG, AGGCTTAG
+#
+gsSamplesheetFileHandle = open(gsSamplesheetFile,'r')
+gsReader = csv.DictReader(gsSamplesheetFileHandle)
+gsHeaders = gsReader.fieldnames
+gsSampleIdColumnName = 'Missing'
+gsSamplesheetDataHashmap = defaultdict(dict)
+gsProjects = []
+if 'Sample_ID' in gsHeaders:
+    gsSampleIdColumnName = 'Sample_ID'
+elif 'ID' in gsHeaders:
+    gsSampleIdColumnName = 'ID'
+else:
+    logging.critical('Cannot find sample ID column name in ' + gsSamplesheetFile + '.')
+    sys.exit('FATAL ERROR!')
+logging.debug('Found sample ID column name ' + gsSampleIdColumnName + ' in ' + gsSamplesheetFile + '.')
+
+for row in gsReader:
+    
+    if row[gsSampleIdColumnName] in (None,""):
+        logging.warning('Empty row detected in GS samplesheet.')
+        continue
+    #
+    # The "sample ID" provided to GenomeScan by our lab is a combination of
+    #  * project
+    #  * sampleProcessStepID; a unique value for each sample.
+    # Use a regular expression to parse the individual values from the combi.
+    #
+    if re.match("(^[a-zA-Z0-9_-]+)-([0-9]+)$", row[gsSampleIdColumnName]):
+        m = re.match("(^[a-zA-Z0-9_-]+)-([0-9]+)$", row[gsSampleIdColumnName])
+        gsProject = m.group(1)
+        gsSampleProcessStepID = m.group(2)
+        logging.debug('Found project ' + gsProject + ' and sampleProcessStepID ' + gsSampleProcessStepID + ' in column ' + gsSampleIdColumnName + '.')
+    else:
+        logging.critical('Cannot parse project name and sampleProcessStepID from "' + row[gsSampleIdColumnName] + '" in column ' + gsSampleIdColumnName + ' from ' + gsSamplesheetFile + '.')
+        sys.exit('FATAL ERROR!')
+    gsGenomeScanID=row['GS_ID']
```

Comment:
I would make this column name a variable just like for ```gsSampleIdColumnName```: would not be surprised if this changes too at some point in time and then we only need to change it once at the top of the script.

Reply:
check, it is specified on line 202 already, i am now using that one